.github/workflows/publish.yml

```diff
@@ -2,13 +2,15 @@ name: Publish
 on:
   release:
     types: [ created ]
```

Comment:
The `publishing.gradle.kts` convention will automatically switch the Maven Central URL to target the main/snapshot repos based on the version in the base `build.gradle.kts` file. If you wanted, we could update this file to automatically publish snapshot releases on any commit to the main branch by triggering this workflow on any master branch commit?

Reply:
> Would it then use the version specified in `build.gradle.kts` and add a `-SNAPSHOT` to it? Or will it get a random "tag" at the end?

I can update the workflow now... but when it runs the Gradle config will use the version in `build.gradle.kts` verbatim, which is 2.3. So unless the version is updated to `2.4-SNAPSHOT` the workflow will try publishing 2.3 (again), not a SNAPSHOT version.

Reply:
Then I wonder how the workflow would look like. Imo `master` should reflect the latest stable version. It is either the published one on Maven or a later version with bugfixes/new features.

A stable master version should get pushed to Maven when we create a new version on github. Like e.g. `2.4`. Ideally, the snapshot versions are also only published from `master`. They could be uploaded manually after each successful commit if the build doesn't break.


How would that work? Or what kind of working do you suggest? Since it is coming from build.gradle.kts, if I understood you correctly, we'd need to update the version between SNAPSHOT and non-SNAPSHOTs? Is it easier to have separate branches?

Reply:
> Then I wonder how the workflow would look like. Imo `master` should reflect the latest stable version. It is either the published one on Maven or a later version with bugfixes/new features.

Yup, I agree, so that's why the version in `build.gradle.kts` should be updated to be a snapshot version that is a preview of the next release (2.4). 

The only git commit that should have a `version = "2.3"` in `build.gradle.kts` should be the commit that's tagged with `v2.3`. The version before this tag should be `2.3-SNAPSHOT` (since that's the next planned release) and the following commits should have a version of `2.4-SNAPSHOT` (since that's the next version that's being previewed). 


This approach is basically [GitHub Flow](https://docs.github.com/en/get-started/quickstart/github-flow).

At the moment it's a little confusing to have `version = "2.3"` in `build.gradle.kts` _after_ v2.3 is already tagged and released - the ship has already sailed! 😄 


Personally I find the whole Git branching and versioning stuff a bit of a headache. I loosely follow GitHub Flow but I wish that it was easier to manage and organise!

Reply:
Okay, so let me phrase it in my own words, if I understood you correctly:
- we now make `2.4-SNAPSHOT`
- when we want to go live with 2.4 we most likely make a 2.4 PR with the changelog e.g. and with the updated version to `2.4`
- when that was committed/merged and successfully published to maven, then we make another commit to `2.5-SNAPSHOT`

Is this correct? If yes, I can live with that! `2.4` will be then also automatically published to maven and is no longer triggered by the Release creation on github repo, correct?

Reply:
yup, that's exactly right!